lib/jekyll/document.rb

```diff
@@ -266,7 +266,7 @@ def read(opts = {})
           merge_defaults
           read_content(opts)
           read_post_data
-        rescue StandardError => e
+        rescue SyntaxError, StandardError, Errors::FatalException => e
```

Comment:
Can you explain what's going on here? Why are we rescuing `FatalException`?

Reply:
Because `:handle_read_error` is:
```ruby
    private
    def handle_read_error(error)
      if error.is_a? SyntaxError
        Jekyll.logger.error "Error:", "YAML Exception reading #{path}: #{error.message}"
      else
        Jekyll.logger.error "Error:", "could not read file #{path}: #{error.message}"
      end

      if site.config["strict_front_matter"] || error.is_a?(Jekyll::Errors::FatalException)
        raise error
      end
    end
```
My understanding is that unless `Errors::FatalException` is rescued here, the variable `e` is not going be a subclass of our `Errors::FatalException`

What say, @parkr ?

Reply:
That doesn't make any sense. We rescue `FatalException` only so that we can rethrow it?

Reply:
Oh, because we need to print the error message about not reading the file. Gotcha 👌

Reply:
Right.. and `Errors::FatalException` is only a subclass of Ruby's `RuntimeError` (which is the default for a bare `raise`), and not a true `fatal` which is documented as `"impossible to rescue"`

Reply:
@pathawks  this change is not actually required for this PR.. do you want me to remove this change so the tests can patched faster?

Reply:
Since `SyntaxError` and `Jekyll::Errors::FatalException` are subclasses of `StandardError`, rescuing from `StandardError` should collect them as well.

For example, in this code:

```ruby
class ASDFError < StandardError; end

def raise_error
  raise ASDFError
end

begin
  raise_error
rescue StandardError => e
  if e.is_a? ASDFError
    puts 'ASDF Error occured'
  else
    puts "Another error occured"
  end
end
```

...rescuing from `StandardError` still collects the `ASDFError`, and logs that an `ASDFError` occurs.